Rearm - Improve framework documentation (#9114)
acemod/ACE3: docs/wiki/framework/rearm-framework.md

</div>

### 1.2 Setting vehicle as a supply

#### 1.2.1 Setting up in the Config

A vehicle will be set as a supply vehicle based on the config `ace_rearm_defaultSupply`
```cpp
class MyTruck: Car_F {
    ace_rearm_defaultSupply = 1200;
};
```

#### 1.2.2 Make an object into a rearm source
*Added in ACE3 3.12.5*

`ace_rearm_fnc_makeSource`

   | Arguments | Type | Optional (default value)
---| --------- | ---- | ------------------------
0  | Rearm Source | Object | Required
1  | Amount | Number | Optional (default: `0`)

> 1  | Add instead set | Bool | Optional (default: false)

Reviewer: `Add instead of set` would be a lot more readable to me.

Author: My initial thought was to keep it short as not to mess with the layout, cause I don't know how to see the exact result for the tables locally. But looking at how other tables handle overflow I changed it to the more verbose `Add amount instead of setting it` and just hope that it will be *fine*.

Author: > Add (`false`) or set (`true`) amount

Would that be an acceptable compromise?

Reviewer: Fine by me

Author: Done

## 3. Functions

#### 3.1 Make an object into a rearm source
*Added in ACE3 3.12.3*

`ace_rearm_fnc_makeSource`

   | Arguments | Type | Optional (default value)
---| --------- | ---- | ------------------------
0  | Ammo Truck | Object | Required
1  | Supply Count | Number | Optional (default: `0`)
1  | Add (`true`) or set (`false`) supply | Bool | Optional (default: `false`)
**R** | None | None | Return value

#### 3.1.1 Example

`[cursorObject, 1200] call ace_rearm_fnc_makeSource`

   | Arguments | Explanation
---| --------- | -----------
0  | `cursorObject` | Rearm source object
1  | `1200` | Ammo supply

### 3.2 Enabling / disabling rearming

`ace_rearm_fnc_disable`

   | Arguments | Type | Optional (default value)
---| --------- | ---- | ------------------------
0  | Ammo Truck | Object | Required
1  | Disable rearming, true to disable, false to enable | Boolean | Optional (default: `true`)
**R** | None | None | Return value

> This functions disables rearming for all supported turrets of a certain vehicle.

Reviewer 2: ```suggestion
This function disables rearming for all supported turrets of a vehicle.
```

#### 3.2.1 Example 1

`[tank] call ace_rearm_fnc_disable;`

   | Arguments | Explanation
---| --------- | -----------
0  |  `tank` | My object

Disables rearming on the object `tank`.

#### 3.2.2 Example 2

`[tank, false] call ace_rearm_fnc_disable;`

   | Arguments | Explanation
---| --------- | -----------
0  |  `tank` | My object
1  |  `false`| Rearming is enabled

Enables rearming on the object `tank`.

### 3.3 Getting the supply count

`ace_rearm_fnc_getSupplyCount`

   | Arguments | Type | Optional (default value)
---| --------- | ---- | ------------------------
0  | Ammo Truck | Object | Required
**R** | Supply count | Number | Return value

> This functions returns the current supply count of the ammo truck.

Reviewer 2: ```suggestion
This function returns the current supply count of the ammo truck.
```

**R** | None | None | Return value

This function is most useful with the module setting `Only specific Magazines`. Note that this function only adds one magazine of a specific class. Other magazines of the same size are not available on this module setting. It has to be used to replenish the ammo truck on `Only specific Magazines` setting.

This function can also be used to increase the supply count on setting `Limited ammo supply based on caliber` by a certain caliber value.

#### 3.5.1 Example

`[ammo_truck, "32Rnd_155mm_Mo_shells"] call ace_rearm_fnc_addMagazineToSupply;`

   | Arguments | Explanation
---| --------- | -----------
0  | `ammo_truck` | My ammo truck object
1  | `"32Rnd_155mm_Mo_shells"` | Some magazine class

The 32 artillery shells are added to the supply count or the magazine storage of the specified ammo truck.

### 3.6 Adding all magazines of a specific vehicle

`ace_rearm_fnc_addVehicleMagazinesToSupply`

   | Arguments | Type | Optional (default value)
---| --------- | ---- | ------------------------
0  | Ammo Truck | Object | Required
1  | Any vehicle object or class name | Object or String | Required
**R** | None | None | Return value

> This functions wraps `ace_rearm_fnc_addMagazineToSupply` and  uses it to add all default magazines of all supported turrets of the vehicle to the ammo truck.

Reviewer 2: ```suggestion
This function wraps `ace_rearm_fnc_addMagazineToSupply` and uses it to add all default magazines of all supported turrets of the vehicle to the ammo truck.
```

0  |  `ammo_truck` | My ammo truck object
1  |  `tank`| A vehicle object

All magazines found in the class config of the object `tank` are made available.

#### 3.6.2 Example 2

`[ammo_truck, "B_MBT_01_arty_F"] call ace_rearm_fnc_addVehicleMagazinesToSupply;`

   | Arguments | Explanation
---| --------- | -----------
0  |  `ammo_truck` | My ammo truck object
1  |  `"B_MBT_01_arty_F"`| Vehicle class name

All magazines found in the config of the vehicle class `B_MBT_01_arty_F` are made available.

### 3.7 Removing magazines from supply

`ace_rearm_fnc_removeMagazineFromSupply`

   | Arguments | Type | Optional (default value)
---| --------- | ---- | ------------------------
0  | Ammo Truck | Object | Required
1  | Magazine Classname | String | Required
2  | Number of Rounds to withdraw | Number | Optional (default: `-1`)
**R** | Magazine could be removed successfully | Boolean | Return value

#### 3.7.1 Example 1

`[ammo_truck, "500Rnd_127x99_mag_Tracer_Red"] call ace_rearm_fnc_removeMagazineFromSupply;`

   | Arguments | Explanation
---| --------- | -----------
0  |  `ammo_truck` | My ammo truck object
1  |  `"500Rnd_127x99_mag_Tracer_Red"`| Carrying is enabled

> Removes one ammo box worth of 500Rnd_127x99_mag_Tracer_Red from the supply. Depending on the module setting the ammo box does hold an entire magazine or only the caliber based amount of rounds.

Reviewer 2: ```suggestion
Removes one ammo box worth of "500Rnd_127x99_mag_Tracer_Red" from the supply. Depending on the module setting the ammo box does hold an entire magazine or only the caliber based amount of rounds.
```

#### 3.7.2 Example 2

`[ammo_truck, "500Rnd_127x99_mag_Tracer_Red", 50] call ace_rearm_fnc_removeMagazineFromSupply;`

   | Arguments | Explanation
---| --------- | -----------
0  |  `ammo_truck` | My ammo truck object
1  |  `"500Rnd_127x99_mag_Tracer_Red"`| Carrying is enabled
2  |  `50` | Number of rounds

> Removes one ammo box with 50 rounds of 500Rnd_127x99_mag_Tracer_Red from the supply. This is 10% of the supply of an entire magazine.

Reviewer 2: ```suggestion
Removes one ammo box with 50 rounds of "500Rnd_127x99_mag_Tracer_Red" from the supply. This is 10% of the supply of an entire magazine.
```
Same with this one.
